### ISSUE TITLE
(UndefinedFunctionError) function X509.PKCS7.sign/3 is undefined (module X509.PKCS7 is not available) after upgrading from x509 0.8.5 => 0.8.6

### ISSUE THREAD

**Issue by neel-desh** [voltone/x509#56]
Hello @voltone , I am facing this issue where after updating x509 0.8.5 => 0.8.6

It it throwing this error while signing

** (UndefinedFunctionError) function X509.PKCS7.sign/3 is undefined (module X509.PKCS7 is not available)

**Comment by voltone** [comment]
That's odd, I'm pretty sure PKCS7 support was never added to X509, so it can't have been removed. I did once [publish](https://elixirforum.com/t/how-can-i-generate-pkcs7-file/46938) a quick-and-dirty sample PKCS7 implementation. Did you perhaps create your own fork of X509 with this module added in?

**Comment by neel-desh** [comment]
No, I never wrote a fork of my own. I did had to refer to the PKCS7 implementation shared by you for reference.

The file just somehow exists in x509
![image](https://user-images.githubusercontent.com/43532747/230338137-e3cd5c02-d03a-4eb8-b627-5364988a46fd.png)

and now I'm worried about where it came from. I'll try to look where it came from. let me know if you have any idea. I can see in git that the file was not added by you. 😟 😨

**Comment by voltone** [comment]
Ah, I suppose I must have had that file in my local directory when publishing 0.8.5 to Hex, and Hex picked it up from there. It was never committed to GitHub, though. The reason I never wanted to make it official is that it would be tricky to support all the various algorithms, as I mentioned in that Elixir Forum post.

Sorry about that! Could you please move the file to your own project once you upgrade to 0.8.6?

**Comment by neel-desh** [comment]
No worries, I had a discussion with my team regarding this. We decided to stick on 0.8.5  for now. 

Thank you soo much tho, for helping with this.

Ps: Should we close the issue then?
